sdk/eidolon_ai_sdk/agent/simple_agent.py

```diff
@@ -73,15 +86,59 @@ def make_output_schema(self, agent, handler):
         return str if self.output_schema == "str" else schema_to_model(self.output_schema, model_name)
 
 
-class SimpleAgentSpec(AgentSpec):
+class NamedCPU(BaseModel):
+    title: Optional[str] = None
+    cpu: AnnotatedReference[AgentCPU]
+    default: bool = False
+
+
+class SimpleAgentSpec(BaseModel):
     description: Optional[str] = None
     system_prompt: str = "You are a helpful assistant"
+    agent_refs: List[str] = []
     actions: List[ActionDefinition] = [ActionDefinition()]
+    cpu: AnnotatedReference[AgentCPU] = None
+    cpus: Optional[List[NamedCPU]] = []
+    can_generate_title: bool = False
```

Comment:
Can you make this title_generation_mode: Literal["none", "on_request"] = "none" ?

The expandable literal will make it nicer if we want to do things like automatically generate the title without user input